sql/core/src/test/resources/sql-tests/inputs/cte.sql

```diff
@@ -155,6 +155,419 @@ SELECT (
+WITH RECURSIVE r(level) AS (
+  VALUES (1L)
+  UNION ALL
+  SELECT SUM(level) FROM r WHERE level < 10
+)
+SELECT * FROM r;
+
+-- aggregate is supported on a recursive table
+WITH RECURSIVE r(level, data) AS (
+  VALUES (0, 'A')
+  UNION ALL
+  SELECT level + 1, data FROM r WHERE level < 10
+)
+SELECT COUNT(*) FROM r;
+
+-- recursive reference is not allowed to use in combination with distinct
+WITH RECURSIVE r(level, data) AS (
+  VALUES (0, 'A')
+  UNION ALL
+  SELECT DISTINCT level + 1, data FROM r WHERE level < 10
+)
+SELECT * FROM r;
+
+-- multiple with works
+WITH RECURSIVE y AS (
+  VALUES (1) AS t(id)
+),
+x AS (
+  SELECT * FROM y
+  UNION ALL
+  SELECT id + 1 FROM x WHERE id < 5
+)
+SELECT * FROM x;
+
+-- multiple with works 2
+WITH RECURSIVE x AS (
+  VALUES (1) AS t(id)
+  UNION ALL
+  SELECT id + 1 FROM x WHERE id < 5
+),
+y AS (
+  VALUES (1) AS t(id)
+  UNION ALL
+  SELECT id + 1 FROM y WHERE id < 10
+)
+SELECT * FROM y LEFT JOIN x ON x.id = y.id;
+
+-- multiple with works 3
+WITH RECURSIVE x AS (
+  VALUES (1) AS t(id)
+  UNION ALL
+  SELECT id + 1 FROM x WHERE id < 5
+),
+y AS (
+  VALUES (1) AS t(id)
+  UNION ALL
+  SELECT id + 1 FROM x WHERE id < 10
+)
+SELECT * FROM y LEFT JOIN x ON x.id = y.id;
+
+-- multiple with works 4
+WITH RECURSIVE x AS (
+  SELECT 1 AS id
+  UNION ALL
+  SELECT id + 1 FROM x WHERE id < 3
+),
+y AS (
+  SELECT * FROM x
+  UNION ALL
+  SELECT * FROM x
+),
+z AS (
+  SELECT * FROM x
+  UNION ALL
+  SELECT id + 1 FROM z WHERE id < 10
+)
+SELECT * FROM z;
+
+-- multiple with works 5
+WITH RECURSIVE x AS (
+  SELECT 1 AS id
+  UNION ALL
+  SELECT id + 1 FROM x WHERE id < 3
+),
+y AS (
+  SELECT * FROM x
+  UNION ALL
+  SELECT * FROM x
+),
+z AS (
+  SELECT * FROM y
+  UNION ALL
+  SELECT id + 1 FROM z WHERE id < 10
+)
+SELECT * FROM z;
+
+-- recursion nested into WITH
+WITH t AS (
+  WITH RECURSIVE s AS (
+    VALUES (1) AS t(i)
+    UNION ALL
+    SELECT i + 1 FROM s
+  )
+  SELECT i AS j FROM s LIMIT 10
+)
+SELECT * FROM t;
+
+-- WITH nested into recursion
+WITH RECURSIVE outermost AS (
+  WITH innermost AS (
+    SELECT * FROM outermost
+  )
+  SELECT level + 1 FROM innermost WHERE level < 5
+  UNION ALL
+  SELECT 0 AS level
+)
+SELECT * FROM outermost;
+
+-- recursion nested into recursion
+WITH RECURSIVE t AS (
+  WITH RECURSIVE s AS (
+    VALUES (1) AS t(i)
+    UNION ALL
+    SELECT i + 1 FROM s WHERE i < 10
+  )
+  SELECT i AS j FROM s
+  UNION ALL
+  SELECT j + 1 FROM t WHERE j < 10
+)
+SELECT * FROM t;
+
+-- recursion nested into recursion 2
+WITH RECURSIVE t AS (
+  WITH RECURSIVE s AS (
+    SELECT j, 1 AS i FROM t
+    UNION ALL
+    SELECT j, i + 1 FROM s WHERE i < 3
+  )
+  VALUES (1) as t(j)
+  UNION ALL
+  SELECT j + 1 FROM s WHERE j < 3
+)
+SELECT * FROM t;
+
+-- routes represented here is as follows:
+--
+-- New York<--->Boston
+-- |            ∧
+-- ∨            |
+-- Washington---+
+-- |
+-- ∨
+-- Raleigh
+CREATE TEMPORARY VIEW routes(origin, destination) AS VALUES
```

Comment:
You forget to drop this view in the end?

Reply:
fixed